erica/erica_legacy/request_processing/requests_controller.py

```diff
@@ -158,6 +164,12 @@ def generate_json(self, pyeric_response: PyericResponse):
 class UnlockCodeActivationRequestController(TransferTicketRequestController):
     _PYERIC_CONTROLLER = UnlockCodeActivationPyericProcessController
 
+    def _is_testmerker_used(self):
+        if self.input_data.tax_id_number:
+            return self.input_data.tax_id_number in SPECIAL_TESTMERKER_IDNR
+        else:
+            return request_needs_testmerker(self.input_data.elster_request_id)
```

Comment:
That is a very different behaviour now, right? If the tax id is given, we check if it's one of the 10, but if it's not given we accept any that starts with 0? I feel like we talked about this, but why not check that it is in SPECIAL_TESTMERKER_IDNR?

Reply:
Very good remark. Somehow my back brain raised that issue today in one of our meetings. I agree with you 👍

Reply:
Moved the implementation into the new tax_id_number_is_test_number function